Added 2 more verification points in CPU offline and vmbus channel interrupt re-assignement

juhlee-microsoft opened the pull request:
1. Assign the random number to the vmbus' cpu id. The random number is in the range of 0 ~ (n-1). n is the max cpu core counts.
2. Assign the original cpu id to the vmbus channel.
Both do not require the VM reboot.

**TEST RESULTS**


[canonical ubuntuserver 16.04-lts Latest] [LISAv2 Test Results Summary]
[canonical ubuntuserver 16.04-lts Latest] Test Run On           : 05/24/2020 19:40:40
[canonical ubuntuserver 16.04-lts Latest] ARM Image Under Test  : canonical : ubuntuserver : 16.04-lts : Latest
[canonical ubuntuserver 16.04-lts Latest] Initial Kernel Version: 4.15.0-1082-azure
[canonical ubuntuserver 16.04-lts Latest] Final Kernel Version  : 5.6.0+
[canonical ubuntuserver 16.04-lts Latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[canonical ubuntuserver 16.04-lts Latest] Total Time (dd:hh:mm) : 0:0:22
[canonical ubuntuserver 16.04-lts Latest] 
[canonical ubuntuserver 16.04-lts Latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[canonical ubuntuserver 16.04-lts Latest] -------------------------------------------------------------------------------------------------------------------------------------------
[canonical ubuntuserver 16.04-lts Latest]     1 CORE                 CPU-OFFLINE-VMBUS-INTERRUPT-REASSINGMENT                                          PASS                 20.2 


[canonical ubuntuserver 18.04-lts Latest] [LISAv2 Test Results Summary]
[canonical ubuntuserver 18.04-lts Latest] Test Run On           : 05/24/2020 19:40:41
[canonical ubuntuserver 18.04-lts Latest] ARM Image Under Test  : canonical : ubuntuserver : 18.04-lts : Latest
[canonical ubuntuserver 18.04-lts Latest] Initial Kernel Version: 5.3.0-1020-azure
[canonical ubuntuserver 18.04-lts Latest] Final Kernel Version  : 5.6.0+
[canonical ubuntuserver 18.04-lts Latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[canonical ubuntuserver 18.04-lts Latest] Total Time (dd:hh:mm) : 0:0:26
[canonical ubuntuserver 18.04-lts Latest] 
[canonical ubuntuserver 18.04-lts Latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[canonical ubuntuserver 18.04-lts Latest] -------------------------------------------------------------------------------------------------------------------------------------------
[canonical ubuntuserver 18.04-lts Latest]     1 CORE                 CPU-OFFLINE-VMBUS-INTERRUPT-REASSINGMENT                                          PASS                23.03 


[canonical ubuntuserver 16.04-lts Latest] [LISAv2 Test Results Summary]
[canonical ubuntuserver 16.04-lts Latest] Test Run On           : 05/24/2020 20:07:28
[canonical ubuntuserver 16.04-lts Latest] ARM Image Under Test  : canonical : ubuntuserver : 16.04-lts : Latest
[canonical ubuntuserver 16.04-lts Latest] Initial Kernel Version: 4.15.0-1082-azure
[canonical ubuntuserver 16.04-lts Latest] Final Kernel Version  : 5.6.0+
[canonical ubuntuserver 16.04-lts Latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[canonical ubuntuserver 16.04-lts Latest] Total Time (dd:hh:mm) : 0:1:26
[canonical ubuntuserver 16.04-lts Latest] 
[canonical ubuntuserver 16.04-lts Latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[canonical ubuntuserver 16.04-lts Latest] -------------------------------------------------------------------------------------------------------------------------------------------
[canonical ubuntuserver 16.04-lts Latest]     1 CORE                 STRESS-CPU-OFFLINE-ONLINE                                                         PASS                84.21 


[canonical ubuntuserver 18.04-lts Latest] [LISAv2 Test Results Summary]
[canonical ubuntuserver 18.04-lts Latest] Test Run On           : 05/24/2020 20:07:29
[canonical ubuntuserver 18.04-lts Latest] ARM Image Under Test  : canonical : ubuntuserver : 18.04-lts : Latest
[canonical ubuntuserver 18.04-lts Latest] Initial Kernel Version: 5.3.0-1020-azure
[canonical ubuntuserver 18.04-lts Latest] Final Kernel Version  : 5.6.0+
[canonical ubuntuserver 18.04-lts Latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[canonical ubuntuserver 18.04-lts Latest] Total Time (dd:hh:mm) : 0:1:30
[canonical ubuntuserver 18.04-lts Latest] 
[canonical ubuntuserver 18.04-lts Latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[canonical ubuntuserver 18.04-lts Latest] -------------------------------------------------------------------------------------------------------------------------------------------
[canonical ubuntuserver 18.04-lts Latest]     1 CORE                 STRESS-CPU-OFFLINE-ONLINE                                                         PASS                87.24

iamshital commented:
Will update the document tomorrow.

lizzha commented on Libraries/CommonFunctions.psm1:
```diff
@@ -639,6 +639,16 @@ function Install-CustomLIS ($CustomLIS, $customLISBranch, $allVMData, [switch]$R
 			$lisSuccess = 0
 			$packageInstallJobs = @()
+			if ($CustomLIS.StartsWith("lisnext") -and (($CustomLIS.Contains("\") -or $CustomLIS.Contains("/")))) {
+				if ($LisBranchProvided) {
```
this variable is not assigned

iamshital replied:
Fixed.

LiliDeng commented on Libraries/CommonFunctions.psm1:
```diff
@@ -639,6 +639,16 @@ function Install-CustomLIS ($CustomLIS, $customLISBranch, $allVMData, [switch]$R
 			$jobCount = 0
 			$lisSuccess = 0
 			$packageInstallJobs = @()
+			if ($CustomLIS.StartsWith("lisnext") -and (($CustomLIS.Contains("\") -or $CustomLIS.Contains("/")))) {
```
$CustomLIS -match "\\|/"

iamshital replied:
Thanks Updated.

LiliDeng commented on Libraries/CommonFunctions.psm1:
```diff
@@ -639,6 +639,16 @@ function Install-CustomLIS ($CustomLIS, $customLISBranch, $allVMData, [switch]$R
 			$packageInstallJobs = @()
+			if ($CustomLIS.StartsWith("lisnext") -and (($CustomLIS.Contains("\") -or $CustomLIS.Contains("/")))) {
+				if ($LisBranchProvided) {
+					$customLISBranch = $CustomLIS.Split("/").Split("\")[1]
```
$CustomLIS.Split("\\|/")

iamshital replied:
Good Suggestion. However, I needed to use `"\\|/"` because `\` is escape character.

LiliDeng replied:
Yes, I copied 
```
\\|/
``` seems also escape is github

iamshital commented:
Execution is verified
-----------------------------
**CustomLIS with branch: .\Run-LisaV2.ps1 -XMLSecretFile 'C:\GitHub\LISAv2\XML\AzureSecrets.xml'  -TestPlatform Azure -TestLocatio
n westus2 -RGIdentifier SS -ARMImageName 'OpenLogic CentOS 7.3 Latest' -TestNames 'BVT-VERIFY-DEPLOYMENT-PROVISION' -ResourceCleanup Delete -EnableTelemetry -CustomLIS 'lisnext\4.3.0'** 

03/05/2019 12:24:46 : [INFO ] Executing customLISInstall.sh ...
03/05/2019 12:24:46 : [INFO ] dos2unix: converting file C:\GitHub\LISAv2\TestResults\2019-05-03-04-21-22-4578\runtest.sh to Unix format...
03/05/2019 12:24:46 : [INFO ] Uploading C:\GitHub\LISAv2\TestResults\2019-05-03-04-21-22-4578\runtest.sh to root : 13.77.177.98, port 22 using Password authentication
03/05/2019 12:24:49 : [INFO ] Upload successful after 1 attempt
03/05/2019 12:24:49 : [INFO ] .\Tools\plink.exe -t -pw ****** -P 22 root@13.77.177.98 "echo ****** | sudo -S /root/customLISInstall.sh -CustomLIS lisnext -LISbranch 4.3.0"
03/05/2019 12:24:53 : [INFO ] /root/customLISInstall.sh -CustomLIS **lisnext** -LISbranch **4.3.0** is running in background with ID 59 ...
.
.
.
03/05/2019 12:35:25 : [INFO ] .\Tools\plink.exe -t -pw ****** -P 22 root@13.77.177.98 "modinfo hv_vmbus"
03/05/2019 12:35:27 : [INFO ] modinfo hv_vmbus executed successfully in 1.95 seconds.
03/05/2019 12:35:27 : [INFO ] Old LIS: filename:       /lib/modules/3.10.0-514.26.2.el7.x86_64/kernel/drivers/hv/hv_vmbus.ko
license:        GPL
rhelversion:    7.3
srcversion:     B18736993819DBB43C5F7B7
alias:          acpi*:VMBus:*
alias:          acpi*:VMBUS:*
depends:
intree:         Y
vermagic:       3.10.0-514.26.2.el7.x86_64 SMP mod_unload modversions
signer:         CentOS Linux kernel signing key
sig_key:        61:8F:5D:DF:77:2E:4B:E8:25:FB:1B:B0:95:91:86:27:24:ED:1E:97
sig_hashalgo:   sha256
03/05/2019 12:35:27 : [INFO ] New LIS: filename:       /lib/modules/3.10.0-514.26.2.el7.x86_64/extra/hv_vmbus.ko
version:        **4.3.0**
license:        GPL
rhelversion:    7.3
srcversion:     467DCAA0A76BDF2D8DF9E75
alias:          acpi*:VMBus:*
alias:          acpi*:VMBUS:*
depends:
vermagic:       3.10.0-514.26.2.el7.x86_64 SMP mod_unload modversions
parm:           affinity_mode:vmbus channel cpu affinity mode: 0, 1 (int)
.
.
.
[LISAv2 Test Results Summary]
Test Run On           : 03/05/2019 12:21:28
ARM Image Under Test  : OpenLogic : CentOS : 7.3 : Latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:21

   ID TestCaseName                                                                TestResult TestDuration(in minutes)
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-DEPLOYMENT-PROVISION                                                   PASS                 2.59
        FirstBoot : PASS
        FirstBoot : Call Trace Verification : PASS
        Reboot : PASS
        Reboot : Call Trace Verification : PASS


Logs can be found at C:\GitHub\LISAv2\TestResults\2019-05-03-04-21-22-4578


03/05/2019 12:43:19 : [INFO ] Current Git Branch:  fix738
03/05/2019 12:43:19 : [INFO ] Current Git Commit ID: f5df75c509b9a311b5ede7601e80706bc46aaa0e
03/05/2019 12:43:19 : [INFO ] Creating 'C:\GitHub\LISAv2\Azure-OL83-TestLogs.zip' from 'C:\GitHub\LISAv2\TestResults\2019-05-03-04-21-22-4578'
03/05/2019 12:43:19 : [INFO ] C:\GitHub\LISAv2\Azure-OL83-TestLogs.zip created successfully.
03/05/2019 12:43:19 : [INFO ] Analyzing results..
03/05/2019 12:43:19 : [INFO ] LISAv2 exit code: 0


**CustomLIS without branch: .\Run-LisaV2.ps1 -XMLSecretFile 'C:\GitHub\LISAv2\XML\AzureSecrets.xml'  -TestPlatform Azure -TestLocation
 westus2 -RGIdentifier SS -ARMImageName 'OpenLogic CentOS 7.3 Latest' -TestNames 'BVT-VERIFY-DEPLOYMENT-PROVISION' -ResourceCleanup Delete -EnableTelemetry -CustomLIS 'lisnext'**

03/05/2019 12:24:21 : [INFO ] **LIS branch was not given. Setting to 'master'**
.
.
.
03/05/2019 12:24:35 : [INFO ] Executing customLISInstall.sh ...
03/05/2019 12:24:35 : [INFO ] dos2unix: converting file C:\GitHub\LISAv2\TestResults\2019-05-03-04-21-09-9577\runtest.sh to Unix format...
03/05/2019 12:24:35 : [INFO ] Uploading C:\GitHub\LISAv2\TestResults\2019-05-03-04-21-09-9577\runtest.sh to root : 13.77.168.41, port 22 using Password authentication
03/05/2019 12:24:37 : [INFO ] Upload successful after 1 attempt
03/05/2019 12:24:37 : [INFO ] .\Tools\plink.exe -t -pw ****** -P 22 root@13.77.168.41 "echo ****** | sudo -S /root/customLISInstall.sh -CustomLIS lisnext -LISbranch master"
03/05/2019 12:24:41 : [INFO ] /root/customLISInstall.sh -CustomLIS lisnext -LISbranch master is running in background with ID 59 ...
.
.
.
03/05/2019 12:29:08 : [INFO ] Old LIS: filename:       /lib/modules/3.10.0-514.26.2.el7.x86_64/kernel/drivers/hv/hv_vmbus.ko
license:        GPL
rhelversion:    7.3
srcversion:     B18736993819DBB43C5F7B7
alias:          acpi*:VMBus:*
alias:          acpi*:VMBUS:*
depends:
intree:         Y
vermagic:       3.10.0-514.26.2.el7.x86_64 SMP mod_unload modversions
signer:         CentOS Linux kernel signing key
sig_key:        61:8F:5D:DF:77:2E:4B:E8:25:FB:1B:B0:95:91:86:27:24:ED:1E:97
sig_hashalgo:   sha256
03/05/2019 12:29:08 : [INFO ] New LIS: filename:       /lib/modules/3.10.0-514.26.2.el7.x86_64/extra/hv_vmbus.ko
version:        **master**
license:        GPL
rhelversion:    7.3
srcversion:     CE2964B6A2652C90F0333C1
alias:          acpi*:VMBus:*
alias:          acpi*:VMBUS:*
depends:
vermagic:       3.10.0-514.26.2.el7.x86_64 SMP mod_unload modversions
parm:           affinity_mode:vmbus channel cpu affinity mode: 0, 1 (int).
.
.
.
[LISAv2 Test Results Summary]
Test Run On           : 03/05/2019 12:21:15
ARM Image Under Test  : OpenLogic : CentOS : 7.3 : Latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:14

   ID TestCaseName                                                                TestResult TestDuration(in minutes)
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-DEPLOYMENT-PROVISION                                                   PASS                 2.86
        FirstBoot : PASS
        FirstBoot : Call Trace Verification : PASS
        Reboot : PASS
        Reboot : Call Trace Verification : PASS


Logs can be found at C:\GitHub\LISAv2\TestResults\2019-05-03-04-21-09-9577


03/05/2019 12:36:01 : [INFO ] Current Git Branch:  fix738
03/05/2019 12:36:01 : [INFO ] Current Git Commit ID: f5df75c509b9a311b5ede7601e80706bc46aaa0e
03/05/2019 12:36:01 : [INFO ] Creating 'C:\GitHub\LISAv2\Azure-OC39-TestLogs.zip' from 'C:\GitHub\LISAv2\TestResults\2019-05-03-04-21-09-9577'
03/05/2019 12:36:01 : [INFO ] C:\GitHub\LISAv2\Azure-OC39-TestLogs.zip created successfully.
03/05/2019 12:36:01 : [INFO ] Analyzing results..
03/05/2019 12:36:01 : [INFO ] LISAv2 exit code: 0

iamshital commented on Libraries/CommonFunctions.psm1:
```diff
@@ -639,6 +639,16 @@ function Install-CustomLIS ($CustomLIS, $customLISBranch, $allVMData, [switch]$R
 			$jobCount = 0
 			$lisSuccess = 0
 			$packageInstallJobs = @()
+			if ($CustomLIS.StartsWith("lisnext") -and (($CustomLIS.Contains("\") -or $CustomLIS.Contains("/")))) {
```
@juhlee-microsoft , CustomLIS parameter not only accepts LIS rpm tarball file, but also it accepts "lisnext" as special string. (This is not new feature. It is already there).
The question Lili put was - it was only taking "master" branch, even though, the shell script had support to clone from other branches.
So I just updated the powershell part to to send LIS branch also along with -CustomLIS parameter.

`-CustomLIS "lisnext"` : will use master branch by default, if no branch is given. (Backward compatibility is unchanged)
`-CustomLIS "lisnext/4.3.0"` : will use 4.3.0 branch.

And, as an added bonus, this will help us to verify the LIS without creating RPM builds.

lizzha commented on Libraries/CommonFunctions.psm1:
```diff
@@ -623,14 +623,14 @@ function Install-CustomKernel ($CustomKernel, $allVMData, [switch]$RestartAfterU
+		if ( !($CustomLIS.StartsWith("lisnext")) -and !($CustomLIS.EndsWith("tar.gz")))
 		{
-			Write-LogErr "Only lisnext and *.tar.gz links are supported. Use -CustomLIS lisnext -LISbranch <branch name>. Or use -CustomLIS <link to tar.gz file>"
+			Write-LogErr "Only lisnext and *.tar.gz links are supported. Use -CustomLIS lisnext/<optinal_branch_name>. Or use -CustomLIS <link to tar.gz file>"
```
typo, optional

juhlee-microsoft committed:
Added new requirement of cpu id change

1. Change to random number to the cpu id
2. Change to the original number to the cpu id

juhlee-microsoft committed:
Fixed the typo in the variable

juhlee-microsoft committed:
Fixed typo

juhlee-microsoft committed:
Verify the change from the reading variable also

juhlee-microsoft committed:
Rebased from master

juhlee-microsoft committed:
Fixed the wording

juhlee-microsoft committed:
Merge branch 'master' into juhlee/random_cpu_id

juhlee-microsoft committed:
Fixed the error

juhlee-microsoft committed:
Update channel_change.sh
